code/modules/vehicles/motorbike.dm

```diff
@@ -0,0 +1,137 @@
+#define FUEL_PER_CAN_POUR 50
+
+/obj/vehicle/ridden/motorbike
+	name = "all-terrain motorbike"
+	desc = "An all-terrain vehicle built for traversing rough terrain with ease. \"TGMC CAVALRY\" is stamped on the side of the engine."
+	icon_state = "motorbike"
+	max_integrity = 300
+	soft_armor = list("melee" = 30, "bullet" = 30, "laser" = 30, "energy" = 0, "bomb" = 30, "fire" = 60, "acid" = 60)
+	obj_flags = CAN_BE_HIT
+	resistance_flags = XENO_DAMAGEABLE
+	key_type = null
+	integrity_failure = 0.5
+	buckle_flags = CAN_BUCKLE|BUCKLE_PREVENTS_PULL|BUCKLE_NEEDS_HAND
+	///Internal motorbick storage object
+	var/obj/item/storage/internal/motorbike_pack/motor_pack = /obj/item/storage/internal/motorbike_pack
+	///Mutable appearance overlay that covers up the mob with th e bike as needed
+	var/mutable_appearance/motorbike_cover
+	///Fuel count, fuel usage is one per tile moved
+	var/fuel_count = 0
+	///max fuel that this bike can hold
+	var/fuel_max = 400
+	COOLDOWN_DECLARE(enginesound_cooldown)
+
+/obj/vehicle/ridden/motorbike/Initialize()
+	. = ..()
+	AddElement(/datum/element/ridable, /datum/component/riding/vehicle/motorbike)
+	motor_pack = new motor_pack(src)
+	motorbike_cover = mutable_appearance(icon, "motorbike_cover", MOB_LAYER + 0.1)
+	fuel_count = fuel_max
```

Comment:
why doing this at the init? So you can change fuel max without having to change fuel count? eh

Reply:
Yes